### PR TITLE
akhil/1340/fetching_the_list_of_wallets_available_to_create

### PR DESCRIPTION
**Pull request by akhil-deriv**
**Clickup link:**  https://app.clickup.com/t/20696747/DERG-1344

This PR contains the following changes:
 - This PR implements the API call that fetches the available wallets to create.

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

## Pre-launch Checklist (For PR creator)

As a creator of this PR:

- [x] ✍️ I have included clickup id and package/app_name in the PR title.
- [x] 👁️ I have gone through the code and removed any temporary changes (commented lines, prints, debug statements etc.).
- [x] ⚒️ I have fixed any errors/warnings shown by the analyzer/linter.
- [x] 📝 I have added documentation, comments and logging wherever required.
- [x] 🧪 I have added necessary tests for these changes.
- [x] 🔎 I have ensured all existing tests are passing.

## Reviewers
@ramin-deriv @waqas-younas-deriv\

## Pre-launch Checklist (For Reviewers)

As a reviewer I ensure that:

- [ ] ✴️ This PR follows the standard PR template.
- [ ] 🪩 The information in this PR properly reflects the code changes.
- [ ] 🧪 All the necessary tests for this PR's are passing.

## Pre-launch Checklist (For QA)

- [ ] 👌 It passes the acceptance criteria.

## Pre-launch Checklist (For Maintainer)

- [ ] [MAINTAINER_NAME] I make sure this PR fulfills its purpose.

### CODE REVIEW

File: lib/basic_api/generated/methods/available_accounts_receive_methods.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+"methods": "",
+"imports": "import 'package:flutter_deriv_api/helpers/helpers.dart';\n"
+}
```
Comment (akhil-deriv) [inline lib/basic_api/generated/methods/available_accounts_receive_methods.json:4]:
```suggestion
}

```

File: lib/api/response/available_accounts_response_extended.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import 'package:flutter_deriv_api/helpers/helpers.dart';
+import 'package:flutter_deriv_api/services/connection/api_manager/base_api.dart';
+
+/// The extended version of the [AvailableAccountsResponse] class to implement
+/// the API call methods.
+class AvailableAccountsResponseExtended extends AvailableAccountsResponse {
+  static final BaseAPI _api = Injector()<BaseAPI>();
+
+  /// Fetches the cashier payments.
+  static Future<AvailableAccountsResponse> fetchCashierPayments({
```
Comment (akhil-deriv) [inline lib/api/response/available_accounts_response_extended.dart:15]:
```suggestion
  static Future<AvailableAccountsResponse> fetchAvailableWalletsToCreate({
```

File: lib/api/response/available_accounts_response_extended.dart [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import 'package:flutter_deriv_api/basic_api/generated/available_accounts_send.dart';
+import 'package:flutter_deriv_api/helpers/helpers.dart';
+import 'package:flutter_deriv_api/services/connection/api_manager/base_api.dart';
+
+/// The extended version of the [AvailableAccountsResponse] class to implement
+/// the API call methods.
+class AvailableAccountsResponseExtended extends AvailableAccountsResponse {
+  static final BaseAPI _api = Injector()<BaseAPI>();
+
+  /// Fetches the cashier payments.
```
Comment (akhil-deriv) [inline lib/api/response/available_accounts_response_extended.dart:14]:
```suggestion
  /// Fetches the available wallets that can be created
```

File: lib/basic_api/helper/response_mapper.helper.dart [hunk 4]
```diff
@@ -222,8 +224,6 @@ Response getGeneratedResponse(Map<String, dynamic> responseMap) {
       return GetAccountStatusReceive.fromJson(responseMap);
     case 'get_account_types':
       return GetAccountTypesReceive.fromJson(responseMap);
-    case 'get_available_accounts_to_transfer':
```
Comment (ramin-deriv) [inline lib/basic_api/helper/response_mapper.helper.dart:225]: I think this call will be used for native cashier. this is removed by the auto generator?

Comment (akhil-deriv) [reply lib/basic_api/helper/response_mapper.helper.dart:225]: Yes, this was removed as part of the auto generation. Do you think I should keep it in?

Comment (ramin-deriv) [reply lib/basic_api/helper/response_mapper.helper.dart:225]: I see, then I think it's fine. 👍

**Review by ramin-deriv — APPROVED** [review]

**Comment by ramin-deriv** [comment]
@akhil-deriv We should switch to `dev` branch. flutter-version-3 has been updated to it recently

**Review by waqas-younas-deriv — APPROVED** [review]

**Pull request merged**